pkg/azuredisk/controllerserver.go

```diff
@@ -524,7 +524,68 @@ func (d *Driver) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (
 
 // ListVolumes return all available volumes
 func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (*csi.ListVolumesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+
+	klog.V(2).Infof("ListVolumes called with request %v", *req)
+
+	var err error
+	start := 0
+	if req.StartingToken != "" {
+		start, err = strconv.Atoi(req.StartingToken)
+		if err != nil {
+			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%s) parsing with error: %v", req.StartingToken, err)
+		}
+		if start < 0 {
+			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) can not be negative", start)
+		}
+	}
+
+	disks, derr := d.cloud.DisksClient.ListByResourceGroup(ctx, d.cloud.ResourceGroup)
+
+	if derr != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Unknown list disks error: %v", derr.Error()))
+	}
+
+	if start >= len(disks) {
+		return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) is greater than total number of volumes", start)
+	}
+
+	pageEnd := len(disks)
+	if req.MaxEntries > 0 {
+		pageEnd = start + int(req.MaxEntries)
```

Comment:
This would allow `pageEnd` to be `> len(disk)` which would cause a run-time panic on line 561 below.
```suggestion
		maxEntries := len(disks) - start
		if maxEntries > int(req.MaxEntries) {
			maxEntries = int(req.MaxEntries)
		}
		pageEnd = start + maxEntries
```